### PR TITLE
docs: Add router mode documentation.

### PR DESCRIPTION
**Pull request by pshelar**
Add router mode related information for AGW user plane.

Signed-off-by: Pravin B Shelar <pbshelar@fb.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: docs/readmes/howtos/ue_ip_address_management.md [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
+# Magma AGW Non-NAT mode
 
-Bridged mode on the Access Gateway refers to any IP allocation strategy that
+Non-NAT mode on the Access Gateway refers to any IP allocation strategy that
 does not NAT assigned UE IP addresses at the AGW.
 
 There are 3 components to reconfigure to enable bridged mode:
 
-- Bridged (Non-NAT) config option for AGW
+- Bridged/Routed (Non-NAT) config option for AGW
```
Comment (electronjoe) [inline docs/readmes/howtos/ue_ip_address_management.md]: Just checking - everywhere else in this PR you moved `Bridged` to `Non-NAT`, but here you moved `Bridged (Non-NAT)` to `Bridged/Routed (Non-NAT)` - just making sure this is clear to the user? The change appears inconsistent to me but may be fine!

**Review by electronjoe — APPROVED** [review]

File: docs/readmes/howtos/ue_ip_address_management.md [hunk 2]
```diff
@@ -20,11 +20,11 @@ update these values if you don't have access to the API. Navigate to the
...
 use the JSON editor to edit its configuration.
 
-## Enable Bridged Mode
+## Enable Non-NAT Mode
 
-To configure an AGW in bridged mode, you need to disable NAT on the AGW.
+To configure an AGW in Non-NAT mode, you need to disable NAT on the AGW.
 Note that this is a gateway-level configuration, so you will need to change
-this for every AGW in the network that you want to run in bridged mode.
+this for every AGW in the network that you want to run in Non-NAT mode.
```
Comment (sudhikan) [inline docs/readmes/howtos/ue_ip_address_management.md:27]: Can we explain what the network wide flag for "NAT" vs "DHCP" does? I think there was some confusion there when we were debugging this with @sourabh-nanoti also.

File: docs/readmes/howtos/ue_ip_address_management.md [hunk 3]
```diff
@@ -39,13 +39,27 @@ This mode will turn on the `uplink_br0` interface in the AGW to bridge UE
...
 node that handles NAT and routing for all user plane traffic. The AGW will
 detect the (default) internet gateway from its networking stack and start ARP
-probes to get the gateway MAC address. This gateway MAC address is used as the
-target host for all bridged user plane traffic. If the AGW is not able complete
-ARP resolution, it floods user plane packets in SGi L2 segment. To avoid
-flooding SGi network, you can statically set the fallback MAC address of the
-internet gateway in `/var/opt/magma/configs/pipelined.yml` with the key
+probes to get the internet gateway MAC address. This gateway MAC address is
+used as the target host for all Non-NAT user plane traffic. If the AGW is not
+able complete ARP resolution, it floods user plane packets in SGi L2 segment.
```
Comment (sudhikan) [inline docs/readmes/howtos/ue_ip_address_management.md]: small nit.

`able to complete` ... `in the SGi L2 segment.` ?

File: docs/readmes/howtos/ue_ip_address_management.md [hunk 4]
```diff
@@ -39,13 +39,27 @@ This mode will turn on the `uplink_br0` interface in the AGW to bridge UE
...
+To avoid flooding SGi network, you can statically set the fallback MAC address
+of the internet gateway in `/var/opt/magma/configs/pipelined.yml` with the key
 `uplink_gw_mac`.
 
+### Router mode
+By default, AGW is in router mode. AGW acts as router for UE subnet. Note, there
```
Comment (sudhikan) [inline docs/readmes/howtos/ue_ip_address_management.md]: `for the UE subnet`.

File: docs/readmes/howtos/ue_ip_address_management.md [hunk 5]
```diff
@@ -39,13 +39,27 @@ This mode will turn on the `uplink_br0` interface in the AGW to bridge UE
...
+is no router advertisement support on AGW, so you need to manually configure the
+SGi IP as next hop for UE IP address subnet(s).
```
Comment (sudhikan) [inline docs/readmes/howtos/ue_ip_address_management.md]: Minor revision:
`
By default, the AGW is pre-configured in `router mode` where the AGW acts as the L3 next hop for the UE allocated subnet (UEs can be allocated IP addresses via the IP address pool or Static assignment).  Note that currently the AGW does not advertise this prefix dynamically so the upstream networks need to be configured with a static route pointing to the SGi interface for the UE subnet(s).
`

File: docs/readmes/howtos/ue_ip_address_management.md (hunk 1 above)
Comment (pshelar) [reply docs/readmes/howtos/ue_ip_address_management.md]: ok, let me fix it.

**Review by sudhikan — CHANGES_REQUESTED** [review]
Some minor comments for you to consider.  Thanks for doing this!

File: docs/readmes/howtos/ue_ip_address_management.md (hunk 2 above)
Comment (pshelar) [reply docs/readmes/howtos/ue_ip_address_management.md:27]: That is explained in the 'UE IP Address Allocation' section of this doc.
I know its confusing, in next release we can add better option for IP allocation in APN API restructuring work.

File: docs/readmes/howtos/ue_ip_address_management.md (hunk 4 above)
Comment (pshelar) [reply docs/readmes/howtos/ue_ip_address_management.md]: there can be multiple UE subnets, fixed it.

**Review by sudhikan — APPROVED** [review]
lgtm

**Pull request merged**